packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -67,6 +67,7 @@ const FormWithRedirect = ({
     sanitizeEmptyValues: shouldSanitizeEmptyValues = true,
     ...props
 }: FormWithRedirectProps) => {
+    const record = useRecordContext(props);
```

Comment:
it feels very weird that you read a context to create it again. 

What I understand is that you want to wrap a RecordContext if a record param is passed. That's what OptionalRecordContextProvider does in next. Maybe the right fix is to use that new component?

Reply:
That would be the right fix but I didn't want to introduce a new component